shared/src/writeToTypegenFile.ts

```diff
@@ -0,0 +1,32 @@
+import * as fs from "fs";
+import * as prettier from "prettier";
+import { promisify } from "util";
+import { getTypegenOutput } from ".";
+
+export const writeToTypegenFile = async (opts: {
+  filePath: string;
+  event: Parameters<typeof getTypegenOutput>[0];
+}) => {
+  const prettierConfig = await prettier.resolveConfig(opts.filePath);
+  const pathToSave = opts.filePath.replace(
+    /\.(mjs|js|tsx|ts|jsx)$/,
```

Comment:
```suggestion
    /\.(cjs|mjs|js|tsx|ts|jsx)$/,
```

Reply:
Although - I'm unsure how this should work for non-TS files, does it even work right now? Technically we could try to process JS files and insert a JSDoc-style type back to the file but I somewhat assume that this is not supported now (and not a high priority for now)